benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/BenchFlowExperimentConfigurationYamlProtocol.scala

```diff
@@ -16,6 +16,10 @@ import scala.util.Try
  */
 object BenchFlowExperimentConfigurationYamlProtocol extends DefaultYamlProtocol {
 
+  // default value since deserializing from a test definition it might not be defined
+  // before use it will be overridden
+  val defaultUsers = 0
```

Comment:
Ok, then you do not need a default. You just need to assign the value you get from the test, and if there is not value as you say it will fail earlier in the lifecycle.